addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4671,6 +4682,183 @@ registry.ReplaceMedia = SnippetOptionWidget.extend({
+     */
+    _computeWidgetState(methodName) {
+        const linkEl = this.$target[0].parentElement.tagName === 'A' ?
+            this.$target[0].parentElement : null;
+        switch (methodName) {
+            case 'setAnchor': {
+                const href = linkEl ? linkEl.getAttribute('href') : null;
+                return href ? '#' + href.split('#')[1] : '';
+            }
+            case 'setLink':
+                return linkEl ? '' : null;
+            case 'setUrl': {
+                const url = this.$urlInputEl.val();
+                const href = linkEl ? linkEl.getAttribute('href') : null;
+                return url || href || '';
+            }
+            case 'setNewWindow': {
+                const target = linkEl ? linkEl.getAttribute('target') : null;
+                return target && target === '_blank' ? '' : null;
+            }
+        }
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    async _computeWidgetVisibility(widgetName) {
+        const linkEl = this.$target[0].parentElement.tagName === 'A' ?
+            this.$target[0].parentElement : null;
+        switch (widgetName) {
+            case 'page_anchor_opt': {
+                const href = linkEl ? linkEl.getAttribute('href') : null;
+                return linkEl && href && href.startsWith('/');
+            }
+            case 'link_on_image_opt':
+                return linkEl;
+        }
+        return this._super(...arguments);
+    },
+    /**
+     * Sets up the display of suggestions available
+     * on the current page or another page of the website.
+     *
+     * @param  {jQueryElement} $inputEl
+     */
+    _setAutoCompletionOnInput($inputEl) {
+        const options = {
+            position: {
+                collision: 'flip fit',
+            },
+            classes: {
+                'ui-autocomplete': 'o_website_ui_autocomplete'
+            },
+        };
+        autocompleteWithPages(this, $inputEl, options);
+    },
+    /**
+     * @override
+     */
+    async _renderCustomXML(uiFragment) {
+        const row = $(uiFragment.querySelector('we-row'));
+        row.append(`<we-button class="fa fa-link" data-set-link="" data-no-preview="true"
+        title="Redirect the user elsewhere when he clicks on the image."></we-button>`);
+        $(`
+        <we-customizeblock-option>
+            <we-input data-set-url="" name="url" data-name="link_on_image_opt"
+            title="Type '/' to search a page. '#' to link to an anchor."
+            string="⌙ Your URL" class="o_we_large" placeholder="www.example.com"/>
+            <we-select class="select_anchor_link" data-name="page_anchor_opt"
+            data-no-preview="true" string="⌙ Page Anchor">
+            </we-select>
+            <we-checkbox data-name="link_on_image_opt" data-no-preview="true"
+            data-set-new-window="" string="⌙ Open in new window"/>
+        </we-customizeblock-option>`).insertAfter(row);
```

Comment:
I believe these will not get properly translated.
Instead of playing with many `_(...)`, maybe putting this in a template and getting it rendered would do the trick.

Reply:
I would not risk XML cache issue in stable IMO, leave it full JS but indeed handle translatable strings. Also, do not use jQuery ever anymore. Especially for this kind of things: standard JS has functions to append/prepend HTML content (which would not force a jQuery parsing etc first).

Reply:
Potentially the static XML could be in a new dedicated file to avoid the cache issue I am talking about, fine for me if you prefer :wink: